src/components/switch/switch.test.tsx

```diff
@@ -0,0 +1,418 @@
+  render(<Switch isOptional label="Optional" />);
+
+  const label = screen.getByText("Optional");
+  const labelContainer = screen.getByTestId("label-container");
+
+  expect(label).toHaveTextContent("Optional");
+  expect(labelContainer).toHaveStyleRule("content", '"(optional)"', {
+    modifier: "::after",
+  });
+});
+
+// Required for coverage
+test("the correct border colour is applied when `error` validation is true", () => {
+  render(<Switch error />);
+
+  const switchSlider = screen.getByTestId("slider");
+
+  expect(switchSlider).toHaveStyleRule(
+    "border-color: var(--colorsSemanticNegative500)"
+  );
+});
+
+// Required for coverage
+test("the correct border colour is applied when `warning` validation is true", () => {
+  render(<Switch warning />);
+
+  const switchSlider = screen.getByTestId("slider");
+
+  expect(switchSlider).toHaveStyleRule(
+    "border-color: var(--colorsSemanticCaution500)"
+  );
+});
+
+// Required for coverage
+test("the correct background colour is applied to the `ErrorBorder` element when `error` validation is a string", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch error="this is an error" />
+    </CarbonProvider>
+  );
+
+  const switchSlider = screen.getByTestId("error-border");
+
+  expect(switchSlider).toHaveStyleRule(
+    "background-color: var(--colorsSemanticNegative500)"
+  );
+});
+
+// Required for coverage
+test("the correct background colour is applied to the `ErrorBorder` element when `warning` validation is a string", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch warning="this is a warning" />
+    </CarbonProvider>
+  );
+
+  const switchSlider = screen.getByTestId("error-border");
+
+  expect(switchSlider).toHaveStyleRule(
+    "background-color: var(--colorsSemanticCaution500)"
+  );
+});
+
+// Required for coverage
+test("renders the expected hint text when `labelHelp` is a string value", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch label="foo" labelHelp="hint text" warning="this is a warning" />
+    </CarbonProvider>
+  );
+
+  expect(screen.getByTestId("hint-text")).toBeVisible();
+});
+
+// Required for coverage
+test("renders the expected margin-bottom value when `labelHelp` is present", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch label="foo" labelHelp="hint text" warning="this is a warning" />
+    </CarbonProvider>
+  );
+
+  const hintTextWrapper = screen.getByTestId("hint-text-wrapper");
+
+  expect(hintTextWrapper).toHaveStyleRule("margin-bottom: var(--spacing000)");
+});
+
+test("the expected translations are correctly applied", () => {
+  render(
+    <I18nProvider
+      locale={{
+        locale: () => "fr-FR",
+        switch: {
+          on: () => "sur",
+          off: () => "de",
+        },
+      }}
+    >
+      <Switch checked onChange={jest.fn()} />
```

Comment:
**nit-pick:** not sure if this was missed but we could also just use a no-op function here 
```suggestion
      <Switch checked onChange={() => {}} />
```